client/my-sites/checkout/checkout-thank-you/domain-mapping-details.jsx

```diff
@@ -10,128 +10,162 @@ import { localize } from 'i18n-calypso';
+			return (
+				<ul className="checkout-thank-you__dns-records-list-placeholder">
+					<li></li>
+					<li></li>
 				</ul>
-			</div>
+			);
+		}
+		const purchasedDomain = getSelectedDomain( { domains, selectedDomainName: domain } );
+		return (
+			<ul className="checkout-thank-you__dns-records-list">
+				{ purchasedDomain.aRecordsRequiredForMapping.map( ( aRecord ) => {
+					return <li key={ aRecord }>{ aRecord }</li>;
+				} ) }
+			</ul>
 		);
-	}
+	};
+
+	const advancedSetupUsingARecordsTitle = translate( 'Advanced setup using root A records' );
+	const aRecordMappingWarning = translate(
+		'If you map a domain using A records rather than WordPress.com name servers, you will need to manage your domain’s DNS records yourself for any other services you are using with your domain, including email forwarding or email hosting (i.e. with Google Workspace or Titan)'
+	);
+	const aRecordsSetupMessage = translate(
+		'Please set the following IP addresses as root A records using {{link}}these instructions{{/link}}:',
+		{
+			components: { link: renderLinkTo( MAP_EXISTING_DOMAIN_UPDATE_A_RECORDS ) },
+		}
+	);
 
-	const description = (
-		<div>
-			{ instructions }
-			<p>{ translate( 'If you already did this, no further action is required.' ) }</p>
+	const renderARecordsMappingMessage = () => {
+		return (
+			<FoldableFAQ id="advanced-mapping-setup" question={ advancedSetupUsingARecordsTitle }>
+				<Notice status="is-warning" showDismiss={ false } translate={ translate }>
+					{ aRecordMappingWarning }
+				</Notice>
+				<p>{ aRecordsSetupMessage }</p>
+				{ renderARecordsList() }
+			</FoldableFAQ>
+		);
+	};
+
+	const purchasedDomain = getSelectedDomain( { domains, selectedDomainName: domain } );
+
+	const renderInstructions = () => (
+		<div className="checkout-thank-you__main-content">
+			{ renderRecommendedSetupMessage( primaryMessage ) }
+			{ purchasedDomain?.aRecordsRequiredForMapping && renderARecordsMappingMessage() }
 		</div>
 	);
+	// TODO: Will remove, added just for the linter to stop complaining
+	renderInstructions();
+
+	const mappingInstructions = (
+		<DomainMappingInstructions
+			isSubdomain={ !! isSubdomainMapping }
+			isLoaded={ isDataLoaded() }
+			domain={ purchasedDomain }
+			translate={ translate }
+		/>
+	);
 
 	return (
 		<div className="checkout-thank-you__domain-mapping-details">
-			<PurchaseDetail
-				icon="cog"
-				description={ description }
-				buttonText={ translate( 'Learn more' ) }
-				href={ isSubdomainMapping ? MAP_SUBDOMAIN : MAP_EXISTING_DOMAIN }
-				target="_blank"
-				rel="noopener noreferrer"
-				isRequired
-			/>
+			<QuerySiteDomains siteId={ siteId } />
+			<PurchaseDetail icon="cog" description={ mappingInstructions } isRequired />
 		</div>
 	);
 };
 
 const mapStateToProps = ( state, { domain } ) => {
 	const selectedSite = getSelectedSite( state );
 	return {
+		domains: getDomainsBySiteId( state, selectedSite.ID ),
 		isBusinessPlan: isBusiness( selectedSite.plan ),
```

Comment:
Probably need to add something like `isRequestingSiteDomains: isRequestingSiteDomains( state, selectedSite.ID )` here from `import { isRequestingSiteDomains } from 'calypso/state/sites/domains/selectors';`.